### ISSUE TITLE
Exception has occurred: OutOfRangeError FIFOQueue '_1_batch_join/fifo_queue' is closed and has insufficient elements (requested 90, current size 0)

### ISSUE THREAD

**Issue by capce** [davidsandberg/facenet#1018]
Hey guys,

I know about the related issues and applied the suggested solutions, but those could not solve the problem.
For training I use about 690.000 aligned .png images with shape 160*160 (some in grayscale) from 9130 classes.

I started the train_softmax application with the following arguments:
`
python src/train_softmax.py --logs_base_dir ./logs/softmax --models_base_dir ./models/softmax_model --data_dir ../vggface2_train_mtcnn_160_reduced/ --image_size 160 --model_def models.inception_resnet_v1 --lfw_dir ../lfw/lfw_mtcnnpy_160 --optimizer ADAM --learning_rate -1 --max_nrof_epochs 150 --keep_probability 0.8 --random_crop --random_flip --use_fixed_image_standardization --learning_rate_schedule_file ./data/learning_rate_schedule_classifier_vggface2.txt --weight_decay 5e-4 --embedding_size 512 --lfw_pairs ./data/pairs.txt --lfw_distance_metric 1 --lfw_use_flipped_images --lfw_subtract_mean --validation_set_split_ratio 0.05 --validate_every_n_epochs 5 --prelogits_norm_loss_factor 5e-4 --gpu_memory_fraction 0.9 --batch_size 90 --epoch_size 1000`

By starting it with those arguments the error occurs at second epoch at batch 566.

Attempts with very small batch_sizes (e.g. 10) and small epoch_size (e.g. 10) were also not successful. The condition batch_size * epoch_size * max_nrof_epochs < total_images * max_nrof_epochs like it is mentioned here https://github.com/davidsandberg/facenet/issues/105 should also be satisfied every time I tried. 
I also checked my dataset for corrupted files with tf.read_file(file) and tf.decode_image(file_contents, 3) but could not find any.

Training was tried on CPU as well as on 3 GTX 1080 TI if that should be relevant. 

Full Error Report:
```
Exception has occurred: OutOfRangeError
FIFOQueue '_1_batch_join/fifo_queue' is closed and has insufficient elements (requested 90, current size 0)
	 [[Node: batch_join = QueueDequeueUpToV2[component_types=[DT_FLOAT, DT_INT32], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](batch_join/fifo_queue, _arg_batch_size_0_0)]]

Caused by op 'batch_join', defined at:
  File "/home/to6ka5/.vscode-remote/extensions/ms-python.python-2019.4.12954/pythonFiles/ptvsd_launcher.py", line 43, in <module>
    main(ptvsdArgs)
  File "/home/to6ka5/.vscode-remote/extensions/ms-python.python-2019.4.12954/pythonFiles/lib/python/ptvsd/__main__.py", line 410, in main
    run()
  File "/home/to6ka5/.vscode-remote/extensions/ms-python.python-2019.4.12954/pythonFiles/lib/python/ptvsd/__main__.py", line 291, in run_file
    runpy.run_path(target, run_name='__main__')
  File "/usr/lib/python3.5/runpy.py", line 254, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/usr/lib/python3.5/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/data/to6ka5/facenet/src/train_softmax.py", line 598, in <module>
    main(parse_arguments(sys.argv[1:]))
  File "/data/to6ka5/facenet/src/train_softmax.py", line 132, in main
    image_batch, label_batch = facenet.create_input_pipeline(input_queue, image_size, nrof_preprocess_threads, batch_size_placeholder)
  File "/data/to6ka5/facenet/src/facenet.py", line 145, in create_input_pipeline
    allow_smaller_final_batch=True)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/training/input.py", line 1148, in batch_join
    name=name)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/training/input.py", line 800, in _batch_join
    dequeued = queue.dequeue_up_to(batch_size, name=name)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/ops/data_flow_ops.py", line 527, in dequeue_up_to
    self._queue_ref, n=n, component_types=self._dtypes, name=name)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 3640, in queue_dequeue_up_to_v2
    component_types=component_types, timeout_ms=timeout_ms, name=name)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 3290, in create_op
    op_def=op_def)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1654, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

OutOfRangeError (see above for traceback): FIFOQueue '_1_batch_join/fifo_queue' is closed and has insufficient elements (requested 90, current size 0)
	 [[Node: batch_join = QueueDequeueUpToV2[component_types=[DT_FLOAT, DT_INT32], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](batch_join/fifo_queue, _arg_batch_size_0_0)]]
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1340, in _do_call
    raise type(e)(node_def, op, message)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1321, in _do_run
    run_metadata)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1140, in _run
    feed_dict_tensor, options, run_metadata)
  File "/home/to6ka5/fn/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 913, in run
    tf_session.TF_DeleteBuffer(options_ptr)
  File "/data/to6ka5/facenet/src/train_softmax.py", line 346, in train
    loss_, _, step_, reg_losses_, prelogits_, cross_entropy_mean_, lr_, prelogits_norm_, accuracy_, center_loss_ = sess.run(tensor_list, feed_dict=feed_dict)
  File "/data/to6ka5/facenet/src/train_softmax.py", line 238, in main
    prelogits, prelogits_center_loss, args.random_rotate, args.random_crop, args.random_flip, prelogits_norm, args.prelogits_hist_max, args.use_fixed_image_standardization)
  File "/data/to6ka5/facenet/src/train_softmax.py", line 598, in <module>
    main(parse_arguments(sys.argv[1:]))
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.5/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.5/runpy.py", line 254, in run_path
    pkg_name=pkg_name, script_name=fname)
```

PS: In Germany we would say "I am at the end of my latin" ;-)

Thank you for advices!

**Comment by xlphs** [comment]
Use opencv imread() or PIL Image.open() to check your training images, I ran into this also and it turned out to be corrupted image files. Here's a script https://gist.github.com/xlphs/2ec4d36b43469fa3ca43b80d96f0ca4f

I think tf doesn't actually *verify* image since it's not an image processing library.

**Comment by cheperuiz** [comment]
I agree with xlphs, I just solved the same issue and was about to post, but @xlphs was 6 minutes ahead of me! 
Cheers

**Comment by capce** [comment]
Thank you both for that hint. I will give it a try and report the result.

**Comment by capce** [comment]
> Use opencv imread() or PIL Image.open() to check your training images, I ran into this also and it turned out to be corrupted image files.

I found one corrupted file, thank you for your help. It seems to work now!
